internal/provider/resource_cloudflare_dlp_profile.go

```diff
@@ -0,0 +1,217 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudflare/cloudflare-go"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareDLPProfile() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareDLPProfileSchema(),
+		CreateContext: resourceCloudflareDLPProfileCreate,
+		ReadContext:   resourceCloudflareDLPProfileRead,
+		UpdateContext: resourceCloudflareDLPProfileUpdate,
+		DeleteContext: resourceCloudflareDLPProfileDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCloudflareDLPProfileImport,
+		},
+		Description: heredoc.Doc(`
+			Provides a Cloudflare DLP Profile resource. Data Loss Prevention profiles 
+			are a set of entries that can be matched in HTTP bodies or files.
+			They are referenced in Zero Trust Gateway rules.
+		`),
+	}
+}
+
+func dlpPatternToSchema(pattern cloudflare.DLPPattern) map[string]interface{} {
+	schema := make(map[string]interface{})
+	if pattern.Regex != "" {
+		schema["regex"] = pattern.Regex
+	}
+	if pattern.Validation != "" {
+		schema["validation"] = pattern.Validation
+	}
+	return schema
+}
+
+func dlpPatternToAPI(pattern map[string]interface{}) cloudflare.DLPPattern {
+	entryPattern := cloudflare.DLPPattern{
+		Regex: pattern["regex"].(string),
+	}
+	if validation, ok := pattern["validation"].(string); ok {
+		entryPattern.Validation = validation
+	}
+	return entryPattern
+}
+
+func dlpEntryToSchema(entry cloudflare.DLPEntry) map[string]interface{} {
+	entrySchema := make(map[string]interface{})
+	if entry.ID != "" {
+		entrySchema["id"] = entry.ID
+	}
+	if entry.Name != "" {
+		entrySchema["name"] = entry.Name
+	}
+	entrySchema["enabled"] = entry.Enabled != nil && *entry.Enabled == true
+	if entry.Pattern != nil {
+		entrySchema["pattern"] = []interface{}{dlpPatternToSchema(*entry.Pattern)}
+	}
+	return entrySchema
+}
+
+func dlpEntryToAPI(entryType string, entryMap map[string]interface{}) cloudflare.DLPEntry {
+	apiEntry := cloudflare.DLPEntry{
+		Name: entryMap["name"].(string),
+	}
+	if entryID, ok := entryMap["id"].(string); ok {
+		apiEntry.ID = entryID
+	}
+	if patterns, ok := entryMap["pattern"].([]interface{}); ok && len(patterns) != 0 {
+		newPattern := dlpPatternToAPI(patterns[0].(map[string]interface{}))
+		apiEntry.Pattern = &newPattern
+	}
+	enabled := entryMap["enabled"] == true
+	apiEntry.Enabled = &enabled
+	apiEntry.Type = entryType
+	return apiEntry
+}
+
+func resourceCloudflareDLPProfileRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+
+	identifier := cloudflare.AccountIdentifier(d.Get("account_id").(string))
+	dlpProfile, err := client.GetDLPProfile(ctx, identifier, d.Id())
+	if err != nil {
```

Comment:
you'll want to be more specific here as any error that bubbles up from `GetDLPProfile` will remove the DLP profile from state.

```suggestion
	var notFoundError *cloudflare.NotFoundError
    if errors.As(err, &notFoundError) {
```